src/ui/components/Timeline/index.tsx

```diff
@@ -120,14 +122,15 @@ export default function Timeline() {
             <Comments />
             <NonLoadingRegions />
             <UnfocusedRegion />
+            {showLoadingProgress && <LoadingProgressBars />}
             <CurrentTimeIndicator editMode={editMode} />
             <Focuser editMode={editMode} setEditMode={setEditMode} />
           </div>
 
           <Tooltip timelineWidth={timelineDimensions.width} />
         </div>
 
-        <Capsule />
+        <Capsule setShowLoadingProgress={setShowLoadingProgress} />
```

Comment:
I'm curious because I haven't used `React.Context` a ton - when do you decide "I should make a context" vs. "I should pass a setter as a prop"? Is there a rule of thumb there? Or is it less about how much you're passing and more about how many consumers there are and how far apart they are in the hierarchy?

Reply:
In general, the Context API exists so that you don't have to do "props drilling" – but there's a small overhead when reading from a context. In this case, since I'm directly rendering a child component, I just chose to pass the value directly as a prop and avoid the second context lookup.

This is pretty subjective and never something I'd block a PR on.